tests/foreach/foreach.t

```diff
@@ -0,0 +1,90 @@
+# Setup few testing env variables
+  $ CRAMPTESTDIR=$PWD
+  $ REPOS=$CRAMPTESTDIR/repos
+
+# Copy all the test directories
+  $ mkdir repos && cd repos && cp -r $TESTDIR/cram_dir/test_dir_* ./
+
+# Edit all urls in subprojects
+  $ cd test_dir_1 &&
+  > sed -i -e "s|PH_TESTDIR2|$REPOS/test_dir_2|g" subprojects.quark &&
+  > sed -i -e "s|PH_TESTDIR3|$REPOS/test_dir_3|g" subprojects.quark &&
+  > cd ..
+  $ cd test_dir_2 &&
+  > sed -i -e "s|PH_TESTDIR4|$REPOS/test_dir_4|g" subprojects.quark &&
+  > cd ..
+
+# Setup the fake git repos
+  $ cd test_dir_1 && git init > /dev/null && git add * && git commit -m "1st commit" > /dev/null && cd ..
+  $ cd test_dir_2 && git init > /dev/null && git add * && git commit -m "1st commit" > /dev/null && cd ..
+  $ cd test_dir_3 && git init > /dev/null && git add * && git commit -m "1st commit" > /dev/null && cd ..
+  $ cd test_dir_4 && git init > /dev/null && git add * && git commit -m "1st commit" > /dev/null && cd ..
+
+# Clone the root testing git repo
+  $ cd $CRAMPTESTDIR
+  $ mkdir checkout && cd checkout
+  $ git clone file://$REPOS/test_dir_1 > /dev/null
+  Cloning into 'test_dir_1'...
```

Comment:
I'm not sure this text is contractual - probably you are better off just ignoring its output.

Reply:
Indeed, on my machine it fails because I have a clone hook that may print stuff.

Reply:
Changed in https://github.com/comelz/quark/pull/27/commits/49652e4a0696c38177c83786a13804b66b91e9f9